examples/keras_io/vision/supervised_contrastive_learning.py

```diff
@@ -0,0 +1,266 @@
+"""
+Title: Supervised Contrastive Learning
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2020/11/30
+Last modified: 2020/11/30
+Description: Using supervised contrastive learning for image classification.
+Accelerator: GPU
+"""
+"""
+## Introduction
+
+[Supervised Contrastive Learning](https://arxiv.org/abs/2004.11362)
+(Prannay Khosla et al.) is a training methodology that outperforms
+supervised training with crossentropy on classification tasks.
+
+Essentially, training an image classification model with Supervised Contrastive
+Learning is performed in two phases:
+
+1. Training an encoder to learn to produce vector representations of input images such
+that representations of images in the same class will be more similar compared to
+representations of images in different classes.
+2. Training a classifier on top of the frozen encoder.
+"""
+"""
+## Setup
+"""
+
+import keras
+from keras import ops
+from keras import layers
+from keras.applications.resnet_v2 import ResNet50V2
+from keras.losses.loss import squeeze_to_same_rank
+import os
+os.environ["KERAS_BACKEND"] = "tensorflow"
```

Comment:
If it works with all backends you can remove this.

Reply:
done